src/open_inwoner/accounts/views/cases.py

```diff
@@ -88,17 +87,31 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
 
         cases = fetch_cases(self.request.user.bsn)
-        case_types = cache.get("case_types")
-        if not case_types:
-            case_types = {case_type.url: case_type for case_type in fetch_case_types()}
-            cache.set("case_types", case_types, 60 * 60)
-        status_types = {
-            status_type.url: status_type for status_type in fetch_status_types()
-        }
-        current_statuses = {
-            status.zaak: status
-            for status in fetch_specific_statuses([case.status for case in cases])
-        }
+
+        case_types = {}
+        status_types = {}
+        temp_status_types = []
+        current_statuses = {}
+
+        for case in cases:
+            # Fetch new case types
+            if case.zaaktype not in case_types.keys():
+                case_type = fetch_single_case_type(case.zaaktype)
+                case_types.update({case_type.url: case_type})
+
+            # Fetch list of case's status types and each one (unique) separately
+            temp_status_types += [st.url for st in fetch_status_types(case.zaaktype)]
+
+            for status_type_url in set(temp_status_types):
+                if status_type_url not in status_types:
+                    status_type = fetch_single_status_type(status_type_url)
+                    status_types.update({status_type.url: status_type})
```

Comment:
Let's discuss what's happening here:

1. For each zaak you fetch statustypes for its zaaktype. 
So it you have 100 cases with 1 zaaktype you will fetch the same statustypen 100 times 

2. For each fetched statustype that you have in `fetch_status_types`, you fetch it again but individually in ` fetch_single_status_type`. Why to do this? They are already fetched, you have all the data. 
So if you have 100 cases with 1 zaaktype, which has 3 statustypen, you will fetch statustypen 100 times and then you will fetch the same statustypen 300 times 

Let's not do this. 
What I suggest:
* don't use `fetch_single_status_type` here, it's excessive
* cache `fetch_status_types`  function for a day

Reply:
This was exactly the opposite thing of optimization.. Apologies for that, I was confused when I was adding the cache decorator.